### PR TITLE
Fix missing Discovery.Azure IExtension

### PR DESCRIPTION
**Pull request by Arkatufus**
Discovery.Azure is missing `AzureDiscovery` class that extends IExtension

## Changes

Add `AzureDiscovery` class

### CODE REVIEW

File: src/discovery/azure/Akka.Discovery.Azure/AkkaHostingExtensions.cs [hunk 1]
```diff
@@ -240,8 +240,12 @@ public static class AkkaHostingExtensions
             AkkaDiscoveryOptions options)
         {
             options.Apply(builder);
-
-            builder.AddHocon(AzureServiceDiscovery.DefaultConfig, HoconAddMode.Append);
+            
+            // force start the module
+            builder.AddStartup((system, registry) =>
+            {
+                AzureDiscovery.Get(system);
```
Comment (Aaronontheweb) [inline src/discovery/azure/Akka.Discovery.Azure/AkkaHostingExtensions.cs:247]: Why is this necessary now but not before?

**Review by Aaronontheweb — COMMENTED** [review]
Just left a question

File: src/discovery/azure/Akka.Discovery.Azure/AkkaHostingExtensions.cs (hunk 1 above)
Comment (Arkatufus) [reply src/discovery/azure/Akka.Discovery.Azure/AkkaHostingExtensions.cs:247]: Before, the discovery module was automatically started by ClusterBootstrap during startup. Since ClusterBootstrap might not run in tandem with AzureDiscovery anymore, we need to force start it.

Comment (Aaronontheweb) [reply src/discovery/azure/Akka.Discovery.Azure/AkkaHostingExtensions.cs:247]: Got it, makes sense

**Review by Aaronontheweb — APPROVED** [review]
LGTM
